pvlib/iotools/cams.py

```diff
@@ -0,0 +1,357 @@
+    mapping.
+
+    See Also
+    --------
+    pvlib.iotools.read_cams_radiation, pvlib.iotools.parse_cams_radiation
+
+    Raises
+    ------
+    requests.HTTPError
+        If the request is invalid, then an XML file is returned by the CAMS
+        service and the error message will be raised as an expcetion.
+
+    References
+    ----------
+    .. [1] `CAMS Radiation Service Info
+       <http://www.soda-pro.com/web-services/radiation/cams-radiation-service/info>`_
+    .. [2] `CAMS McClear Service Info
+       <http://www.soda-pro.com/web-services/radiation/cams-mcclear/info>`_
+    .. [3] `CAMS McClear Automatic Access
+       <http://www.soda-pro.com/help/cams-services/cams-mcclear-service/automatic-access>`_
+    """
+    if time_step in TIME_STEPS_MAP.keys():
+        time_step_str = TIME_STEPS_MAP[time_step]
+    else:
+        warnings.warn('Time step not recognized, 1 hour time step used!')
+        time_step, time_step_str = '1h', 'PT01H'
+
+    if (verbose is True) & ((time_step != '1min') | (time_ref != 'UT')):
+        verbose = False
+        warnings.warn("Verbose mode only supports 1 min. UT time series!")
+
+    # Format verbose variable to the required format: {'true', 'false'}
+    verbose = str(verbose).lower()
+
+    if altitude is None:  # Let SoDa get elevation from the NASA SRTM database
+        altitude = -999
+
+    # Start and end date should be in the format: yyyy-mm-dd
+    start_date = start_date.strftime('%Y-%m-%d')
+    end_date = end_date.strftime('%Y-%m-%d')
+
+    email = email.replace('@', '%2540')  # Format email address
+    service = 'get_{}'.format(service.lower())  # Format CAMS service string
+
+    # Manual format the request url, due to uncommon usage of & and ; in url
+    url = ("http://{}/service/wps?Service=WPS&Request=Execute&"
+           "Identifier={}&version=1.0.0&RawDataOutput=irradiation&"
+           "DataInputs=latitude={};longitude={};altitude={};"
+           "date_begin={};date_end={};time_ref={};summarization={};"
+           "username={};verbose={}"
+           ).format(server, service, latitude, longitude, altitude, start_date,
+                    end_date, time_ref, time_step_str, email, verbose)
+
+    res = requests.get(url)
+
+    # Invalid requests returns helpful XML error message
+    if res.headers['Content-Type'] == 'application/xml':
+        errors = res.text.split('ows:ExceptionText')[1][1:-2]
+        raise requests.HTTPError(errors, response=res)
+    # Check if returned file is a csv data file
+    elif res.headers['Content-Type'] == 'application/csv':
+        fbuf = io.StringIO(res.content.decode('utf-8'))
+        data, meta = parse_cams_radiation(fbuf, integrated=integrated,
+                                          label=label,
+                                          map_variables=map_variables)
+        return data, meta
+    else:
+        warnings.warn('File content type not recognized.')
```

Comment:
I would hope that it's not possible for a user to get here and we don't need it. If we do need it then it should be an exception instead of a warning.

Reply:
Ok, I will remove this.